RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -500,9 +707,130 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
 	      }
           } // end of HBHE check
      } // loop on HcalCalibDigiCollection
+
+     // now match the laser monitor data by fiber (in time) 
+     if( fillLaserMonitor_ ) {
+       // check for any fibers without data and fill
+       // them so we dont run into problems later
+       for( unsigned idx = 0; idx < laserMonDetTypeList_.size(); ++idx ) {
+           if( lasmon_adcs[idx].empty() ) {
+               lasmon_adcs[idx] = std::vector<int>(10, -1);
+           }
+           if( lasmon_capids[idx].empty() ) {
+               lasmon_capids[idx] = std::vector<int>(10, -1);
+           }
+       }
+       unsigned nFibers = laserMonIEtaList_.size();
+       // for each fiber we need to find the index at with the 
+       // data from the next fiber matches in order to stitch them together.
+       // When there is an overlap, the data from the end of the
+       // earlier fiber is removed.  There is no removal of the last fiber
+       std::vector<unsigned> matching_idx; 
+       // we assume that the list of fibers was given in time order
+       // (if this was not the case, then we just end up using 
+       // all data from all fibers )
+       for( unsigned fidx = 0; fidx < (nFibers - 1); ++fidx ) {
+         // start with the last TS and loop backwards
+         // on each iteration check if all capId and ADCs from this
+         // TS forward match the beginning entries of 
+         // the next fiber
+         int last_ts = lasmon_capids[fidx].size()-1;  // last TS
+         int start_ts = last_ts; // start_ts will be decrimented on each loop
+         // in the case that our stringent check below doesn't work 
+         // store the latest capID that has a match
+         int latest_cap_match = -1;
+         // reverse loop over TSs
+         while( start_ts >= 0 ) {
+           int ncheck = 0; // count the number of TS that were checked
+           int nmatch_cap = 0; // count the number of TS where capID matched
+           int nmatch_adc = 0; // count the number of TS where ADC matched
+
+           for( int cidx = start_ts, nidx = 0; cidx <= last_ts; cidx++, nidx++ ) { 
+             ncheck++;
+             // if we get an invald value, move on
+             if( lasmon_capids[fidx][cidx] == -1 ) continue;
```

Comment:
You can just break here: if( lasmon_capids[fidx][cidx] == -1 )  there is no possibility any more that nmatch_cap and nmatch_adc can become equal to ncheck

Reply:
yes, done